src/wallet/wallet.cpp

```diff
@@ -3273,6 +3273,18 @@ static void ApproximateBestSubset(vector<pair<CAmount, pair<const CWalletTx*,uns
     }
 }
 
+// std::random_shuffle is removed in C++17.
```

Comment:
`std::random_shuffle` was removed in C++17 because `std::shuffle` was added in C++11. Is there any reason to not just switch to `std::shuffle`?

Reply:
`std::shuffle` requires a generator class with a specific interface, which I'm adding to `src/random.h`.